stubs/docutils/docutils/parsers/rst/__init__.pyi

```diff
@@ -1,34 +1,66 @@
-from _typeshed import Incomplete
+from collections.abc import Callable, Sequence
 from typing import Any, ClassVar, Literal
+from typing_extensions import TypeAlias
 
-from docutils import parsers
-from docutils.parsers.rst import states
+from docutils import nodes, parsers
+from docutils.parsers.rst.states import Inliner, RSTState, RSTStateMachine
+from docutils.statemachine import StringList
+from docutils.transforms import Transform
 
 class Parser(parsers.Parser):
+    settings_spec: ClassVar[tuple[str, str | None, tuple[tuple[str, list[str], dict[str, Any]], ...]]]
```

Comment:
@picnixz any ideas here?

Reply:
See `docutils/__init__.py` which gives you an explanation about that object. It essentially tells you the configuration of the argument parser (which options, what to do, what they do, etc). Best idea is just to ignore literal values since it can easily change and/or you write a mypy plugin for type checking this (but honestly, it'll be hard to maintain).

Reply:
so putting aside the literal strings, what is the signature?
I can match the runtime signature exactly (see https://github.com/python/typeshed/pull/11523#issuecomment-1975190781) but that's just one instance of this spec- what is the actual structure?